### PR TITLE
proc: implement attach on windows

### PR DESCRIPTION
**Pull request by alexbrainman**
This change implements attach command on windows

Updates #363

---
This change is [Reviewable](https://reviewable.io/reviews/derekparker/delve/607)

### CODE REVIEW

**Comment by derekparker** [comment]
Awesome, thanks for this! I'll do a full review sometime this afternoon, in the meantime @lukehoban mind reading through this?

File: proc/proc_windows.go [hunk 1]
```diff
@@ -126,13 +121,56 @@ func Launch(cmd []string) (*Process, error) {
...
+}
+
+// findExePath searches for process pid, and returns its executable path.
+func findExePath(pid int) (string, error) {
+	// Original code suggested different approach (see below).
+	// Maybe it could be useful in the future.
+	//
+	// Find executable path from PID/handle on Windows:
+	// https://msdn.microsoft.com/en-us/library/aa366789(VS.85).aspx
+	//
```
Comment (derekparker) [inline proc/proc_windows.go]: nit: remove this line.

File: proc/proc_windows.go [hunk 2]
```diff
@@ -126,13 +121,56 @@ func Launch(cmd []string) (*Process, error) {
...
+		default:
+			return "", err
+		}
+	}
 }
 
 // Attach to an existing process with the given PID.
 func Attach(pid int) (*Process, error) {
-	return nil, fmt.Errorf("not implemented: Attach")
+	// TODO: Probably should have SeDebugPrivilege before starting here.
```
Comment (derekparker) [inline proc/proc_windows.go]: Does this apply to the exec path as well? Is it possible we can check for that, and return an appropriate error?

**Comment by derekparker** [comment]
Overall looks good, just a nit and question. Thanks for this!

File: proc/proc_windows.go (hunk 2 above)
Comment (alexbrainman) [inline proc/proc_windows.go]: I am not clear what you are asking about. If you asking if we need to hold SeDebugPrivilege when we are starting debugged process ourselves (as well as when attaching to external process), then I do not know. CreateProcess Windows API does not mention SeDebugPrivilege, while DebugActiveProcess does.

I played around with new "attach" functionality, and I can attach to external process (small Go program) with code in this PR. So it seems I don't always need to ask for SeDebugPrivilege. But I have no doubt we will find ourselves in a situation when this is required. So I have added a TODO, so we don't forget. I suspect we could try acquire SeDebugPrivilege once at the start of the program, and continue with our job regardless of result - maybe display warning message if SeDebugPrivilege could not be acquired.

File: proc/proc_windows.go (hunk 1 above)
Comment (alexbrainman) [inline proc/proc_windows.go]: I have just done that. I have changed my local commit and pushed it (with -f) into my fork https://github.com/alexbrainman/delve/commit/f09ef23f9ec650f18f2530d7164229ea4a119700 I don't understand why this pull request does not see my change. I don't use PRs. Please help.

**Pull request merged**